Dockerfile

```diff
@@ -0,0 +1,41 @@
+# fog-sw BUILDER
+ARG ROS_DISTRO="galactic"
+FROM ros:${ROS_DISTRO}-ros-base as fog-sw-builder
+
+ARG UID=1000
+ARG GID=1000
+ARG BUILD_NUMBER
+ARG COMMIT_ID
+ARG GIT_VER
+ARG PACKAGE_NAME
+# Install build dependencies
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
+    curl \
+    python3-bloom \
+    fakeroot \
+    dh-make \
+    libboost-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN groupadd -g $GID builder && \
+    useradd -m -u $UID -g $GID -g builder builder && \
+    usermod -aG sudo builder && \
+    echo 'builder ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+
+RUN echo "deb [trusted=yes] https://ssrc.jfrog.io/artifactory/ssrc-debian-public-remote focal fog-sw" >> /etc/apt/sources.list
```

Comment:
why is this needed here?

Reply:
Not needed for this package. We created some template scripts that can be used for other packages as well. For instance, the fog_bumper would have fog-msgs dependency, so it can get from the artifactory repo. 

The scripts also have ability to fetch dependencies using underlay.repos if the requested version is not available in the artifactory using vcs tool.

Reply:
The example for this dependency gathering can be found in [odometry2/DP-1761-new_build_process](https://github.com/tiiuae/odometry2/tree/DP-1761-new_build_process). 

If the requested version does not exist in the artifactory, it will get the package using underlay.repos, build it and create the deb.